cellpack/autopack/Ingredient.py

```diff
@@ -3241,86 +3340,81 @@ def checkSphCollisions(
                     self.vi.updateBox(box, cornerPoints=bb)
                 self.vi.update()
 
-            pointsInCube = histoVol.grid.getPointsInCube(
-                bb, posc, radius_of_area_to_check, info=True
+            pointsToCheck = histoVol.grid.getPointsInSphere(
+                posc, radius_of_area_to_check
             )  # indices
             # check for collisions by looking at grid points in the sphere of radius radc
-            delta = numpy.take(gridPointsCoords, pointsInCube, 0) - posc
+            delta = numpy.take(gridPointsCoords, pointsToCheck, 0) - posc
             delta *= delta
             distA = numpy.sqrt(delta.sum(1))
-            # loop over all the grid points in the cube
-            # NOTE: used to filter these points to an inscribed sphere, but is expensive and didn't reduce points to check
-            for pti in range(len(pointsInCube)):
-                pt = pointsInCube[
+
+            for pti in range(len(pointsToCheck)):
+                pt = pointsToCheck[
                     pti
                 ]  # index of master grid point that is inside the sphere
                 distance_to_packing_location = distA[
                     pti
                 ]  # is that point's distance from the center of the sphere (packing location)
-
                 # distance is an array of distance of closest contact to anything currently in the grid
-                if (
-                    distance_to_packing_location <= radius_of_ing_being_packed
-                    and distance[pt] < -0.0001
-                ):
-                    # if there is an object encroaching into this packing area, that is
-                    # closer to the packing location than the radius of the object
-                    # we are trying to pack, probably a collision
-                    if level < self.maxLevel and (level + 1) < len(self.positions):
-                        nxtLevelSpheres = self.positions[level + 1]
-                        nxtLevelRadii = self.radii[level + 1]
+                collision = (
+                    distance[pt] + distance_to_packing_location
+                    <= radius_of_ing_being_packed
```

Comment:
This is the change that actually prevents collisions. We need to use the distance to the closes surface to determine if the closes surface to too close to the packing location we're trying

Reply:
do we want to have the other check as a "soft" packing option?